contracts/axelarnet-gateway/src/msg.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::HexBinary;
+use msgs_derive::EnsurePermissions;
+use router_api::{Address, ChainName, CrossChainId, Message};
+
+use crate::state::MessageWithStatus;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// The chain name for this gateway.
+    pub chain_name: ChainName,
+    /// Address of the router contract on axelar.
+    pub router_address: String,
+}
+
+#[cw_serde]
+#[derive(EnsurePermissions)]
+pub enum ExecuteMsg {
+    /// Initiate a cross-chain contract call from Axelarnet to another chain.
+    /// The message will be routed to the destination chain's gateway via the router.
+    #[permission(Any)]
+    CallContract {
+        destination_chain: ChainName,
+        destination_address: Address,
+        payload: HexBinary,
+    },
+
+    /// Forward the given messages to the next step of the routing layer.
+    /// Messages initiated via `CallContract` can be forwarded again to the router.
+    /// If the messages are coming from the router, then they are marked ready for execution.
+    #[permission(Any)]
+    RouteMessages(Vec<Message>),
+
+    /// Execute the message at the destination contract with the corresponding payload.
+    #[permission(Any)]
+    Execute {
+        cc_id: CrossChainId,
+        payload: HexBinary,
+    },
+}
+
+#[cw_serde]
+#[derive(QueryResponses)]
+pub enum QueryMsg {
+    /// Returns the list of messages with their status destined for Axelar.
+    #[returns(Vec<MessageWithStatus>)]
+    OutgoingMessages { message_ids: Vec<CrossChainId> },
```

Comment:
Probably would be useful to have a query for the messages initiated via CallContract as well, though we could add in a separate PR, since this is already quite large.

Also, I find the nomenclature confusing here. For the other gateways, we call the messages returned by this query "outgoing messages", and it makes sense because those messages are going to be relayed from the gateway on axelar to a gateway on another chain (those messages are leaving axelar, in a sense). Versus the messages relayed from the source to axelar are incoming. But in this case, I feel like it's more natural to call the messages initiated via CallContract as outgoing (they will leave axelar), and messages received from the router as incoming (they will be executed on axelar). 

I don't want to switch the standard nomenclature we've been using, because I feel like that would also be confusing. I think you could just call this query `GetMessages` and add a status for messages initiated via CallContract ('Sent'?).

Reply:
Hmm, I wanted to preserve the API that other gateways expose (another name will also break it). Agreed that it's confusing, although not sure about calling it just `GetMessages`

Reply:
Well the return type is already different, so the API is already not preserved. For normal gateways, this is also mainly used by the prover, but there is no prover in this case. Furthermore, this gateway is different than the others, since it's not an intermediate gateway. So I would rather this gateway have an API that is clear and understandable based on what it does, rather than try to conform to what the other gateways do and be confusing.

Reply:
I also found incoming / outgoing confusing. In the context of gateway itself,
- messages from router are incoming
- call contract is outgoing

I am open to name it more explicitly

Reply:
I renamed it to `send_messages` and `receive_messages`

Reply:
could you also rename in state file